tasks/ast_tools/src/main.rs

```diff
@@ -128,5 +137,41 @@ fn write_ci_filter(
     push_item("tasks/ast_codegen/src/**/*");
     push_item(output_path);
 
-    write_all_to(output.as_bytes(), output_path)
+    log!("Writing {output_path}...");
+    write_all_to(output.as_bytes(), output_path)?;
+    logln!(" Done!");
+    Ok(())
 }
+
+#[macro_use]
+mod logger {
+    static mut LOG: bool = true;
+
+    /// Shouldn't be called after first use of the logger macros.
+    pub(super) unsafe fn quiet() {
+        LOG = false;
+    }
+
+    pub(super) fn __internal_log_enable() -> bool {
+        // SAFETY:`LOG` doesn't change from the moment we start using it.
+        unsafe { LOG }
+    }
```

Comment:
Once again, I'm late to the party!

Do we need to resort to unsafe code for this? Could we use something like a `OnceLock` or `AtomicBool` instead?

I know that'd be a bit less efficient, but I imagine it'd be a drop in the ocean in the context of the whole codegen process, and performance is not critical in AST tools anyway.

Reply:
You are correct, I'll submit a PR for it.

Reply:
#5574 will fix this